apps/src/templates/EligibilityConfirmDialog.jsx

```diff
@@ -0,0 +1,85 @@
+/** @file Confirm Dialog for Maker Discount Codes */
+import React, {Component} from 'react';
+import i18n from "@cdo/locale";
+import BaseDialog from "./BaseDialog";
+import DialogFooter from "./teacherDashboard/DialogFooter";
+import Button from "./Button";
+
+const styles = {
+  subtitle: {
+    fontSize: 16,
+  }
+};
+
+export default class EligibilityConfirmDialog extends Component {
+  state = {
+    checkStudentCount: false,
+    checkYear: false,
+    checkSingleCode: false,
+    signature: "",
+    activeButton: false,
+  };
+
+  verifyResponse = () => {
+    if (this.state.checkStudentCount &&
+        this.state.checkYear &&
+        this.state.checkSingleCode &&
+        this.state.signature !== ""){
+      this.setState({activeButton : true});
+    } else { // if someone unchecks their responses
+      this.setState({activeButton: false});
+    }
+  }
+
+  setSignature = (value) => {
+    this.setState({signature : value});
+    this.verifyResponse();
+  }
+
+  render() {
+    return (
+      <BaseDialog
+        useUpdatedStyles
+        uncloseable
+        isOpen = {true}
+        assetUrl={() => ''}
+        style={{padding:20}}
+      >
+        <h2>{i18n.getCode()}</h2>
+        <div style={styles.subtitle} >{i18n.verifyStatementsforCode()}</div>
+        <br/>
+        <form>
+          <label style={styles.boxStyle}>
+            <input type="checkbox" value="verify1" onChange={() => {this.setState({checkStudentCount : !this.state.checkStudentCount}); this.verifyResponse();}}/>
```

Comment:
In general we should avoid having inlined functions in our render function.  Except in cases where attributes are very short, we should also have one attribute per line.